### DISCUSSION TITLE
Request for new keyboard support

### DISCUSSION THREAD

**Discussion by cfwh** [manna-harbour/miryoku#232] [Q&A] [answered]
I've recently finished my setting up ZMK for my keyboard, the OTG, and would like to try it out using Miryoku too.  How would I get my shield added to Miryoku ZMK?

**Comment by manna-harbour** [comment]
The process of adding support for new keyboards isn't documented, but if you'd like to do it yourself there are plenty of examples in the commit history. Otherwise I'm happy to do it.

**Reply by cfwh** [reply]
That'd be very helpful thank you. Once done, I can use Github actions as documented in the [quickstart guide](https://github.com/manna-harbour/miryoku_zmk/tree/master/docs/quickstart) yeah?

**Reply by manna-harbour** [reply]
Sure, no problem!

Yes, after it's added it'll be used automatically for any workflow builds for that keyboard. If building locally you'll need to manually add the files in the appropriate location.

**Reply by cfwh** [reply]
Thanks. I don't think I'll be touching local builds, I barely got through Actions

**Reply by manna-harbour** [reply]
Ok, here's a sample build. Let me know if it seems ok and I'll push the changes.
[miryoku_zmk-OTG-nice_nano_v2.zip](https://github.com/manna-harbour/miryoku/files/12266855/miryoku_zmk-OTG-nice_nano_v2.zip)

**Reply by cfwh** [reply]
Everything seems to work except undo and redo, the toggle stuff on the media layer because I don't know what it does, mouse because it shouldn't, the rgb stuff on the media layer because I don't have RGB, and i'll test the layer locks and bootloader when I get back

**Reply by manna-harbour** [reply]
If the keys are in the right places then everything else should be fine.

**Reply by cfwh** [reply]
Oh excellent. I figured out the toggle stuff and the bootloader works but I still can't get undo and redo to work and am not entirely sure how the layer locks work

**Comment by manna-harbour** [comment]
ZMK keyboard names seem to be all lowercase, so you may want to rename file name and contents.

**Reply by manna-harbour** [reply]
If you do decide to rename, test then let me know and I'll update.

**Reply by cfwh** [reply]
Is this a standardisation thing?

**Reply by manna-harbour** [reply]
It doesn't matter for Miryoku ZMK, and if it works I can add it as-is. But if you want to PR to ZMK, or fit in with their eventual out-of-tree metadata directory, then you'll need to meet their standards. The convention seems to be lowercase alphanumeric plus underscore only. I don't see it covered in the ZMK docs, so you might like to ask ZMK.

**Reply by cfwh** [reply]
Convention! That was the word I was looking for. What benefits are there to PR-ing to ZMK or fitting with their eventual out-of-tree metadata directory (whatever that means)?

**Reply by manna-harbour** [reply]
Probably none in practice, as ZMK tend to discourage PRs, and there's no firm plans yet for the directory. It's more a general issue of going with convention, and potentially avoiding compatibility issues in future.

**Reply by cfwh** [reply]
hey sorry it took so long but I've lowercased everything and it works!

**Comment by manna-harbour** [comment, marked as answer]
Added in https://github.com/manna-harbour/miryoku_zmk/commit/d43e5ddfdcb73e346c8a39173bf63ebbb969a867. Here's a sample build run: https://github.com/manna-harbour/miryoku_zmk/actions/runs/6172043351.
